src/reactor/cache.js

```diff
@@ -0,0 +1,201 @@
+import { List, Map, OrderedSet, Record } from 'immutable'
+
+export const CacheEntry = Record({
+  value: null,
+  storeStates: Map(),
+  dispatchId: null,
+})
+
+/*******************************************************************************
+ * interface PersistentCache {
+ *    has(item)
```

Comment:
My initial approach had this, but changing from a immutable to mutable cache had negative implications (e.g. when [reactor evaluates using previous state](https://github.com/optimizely/nuclear-js/blob/1444ce6b89c91031c989ff07cf168e630facb00a/src/reactor.js#L241) causes churn)